components/utils/src/fs.rs

```diff
@@ -106,7 +107,11 @@ pub fn copy_file(src: &Path, dest: &PathBuf, base_path: &PathBuf, hard_link: boo
     if hard_link {
         std::fs::hard_link(src, target_path)?
     } else {
-        copy(src, target_path)?;
+        copy(src, &target_path)?;
+        // set original modification timestamp to the target file
+        let src_metadata = metadata(src)?;
+        let mtime = FileTime::from_last_modification_time(&src_metadata);
+        set_file_mtime(&target_path, mtime)?;
```

Comment:
Can we avoid copying the file if it exists and the mtime is identical to the src?

Reply:
Sure, that sounds a nice idea. I'll implement it.

Reply:
I've done it :)

Reply:
> Can we avoid copying the file if it exists and the mtime is identical to the src?

Need to also compare SIZE in addition to the mtime.

Reply:
You're exactly right. Like `rsync` command, the decision whether copy operation is necessary or not should be done by comparing both timestamp and filesize of two files.
cf. https://stackoverflow.com/questions/13778889/rsync-difference-between-size-only-and-ignore-times/44353773
I'll write it. Thanks.

Reply:
Added it!